src/session_impl.cpp

```diff
@@ -1685,6 +1689,8 @@ namespace aux {
 					!= m_settings.get_str(settings_pack::listen_interfaces));
 
 		apply_pack(&pack, m_settings, this);
+		INVARIANT_CHECK;
```

Comment:
I think it would be better to place this at the very top of the function. This macro will cause the invariant check to be run twice, once immediately and once at the end of the scope (in this case when the function returns)